store/tikv/txn.go

```diff
@@ -428,13 +429,17 @@ func (txn *tikvTxn) LockKeys(ctx context.Context, forUpdateTS uint64, keysInput
 			for _, key := range keys {
 				txn.us.DeleteKeyExistErrInfo(key)
 			}
-			wg := txn.asyncPessimisticRollback(ctx, keys)
-			if dl, ok := errors.Cause(err).(*ErrDeadlock); ok && hashInKeys(dl.DeadlockKeyHash, keys) {
-				dl.IsRetryable = true
-				// Wait for the pessimistic rollback to finish before we retry the statement.
-				wg.Wait()
-				// Sleep a little, wait for the other transaction that blocked by this transaction to acquire the lock.
-				time.Sleep(time.Millisecond * 5)
+			keyMayBeLocked := terror.ErrorNotEqual(kv.ErrWriteConflict, err) && terror.ErrorNotEqual(kv.ErrKeyExists, err)
```

Comment:
I'm afraid if we add more error types or changes our usage of these error types in the future, this may fail.

Reply:
That is why it use `ErrorNotEqual` here. As long as we make sure `ErrWriteConflict` and `ErrKeyExists` won't lock the key.